### PR TITLE
Docs: Row/column hiding

### LINKED ISSUE #7390: HiddenRows documentation clarification: does it still use height: 0px?
### Description
This is a question, not a bug / feature request.

The [documentation for the `HiddenRows` plugin](https://handsontable.com/docs/8.2.0/HiddenRows.html) says:

> Plugin allows to hide certain rows. The hiding is achieved by rendering the rows with height set as 0px.

Is this still true after the 8.0.0 refactoring and the addition of `rowIndexMapper`? In particular, https://github.com/handsontable/handsontable/commit/2796dc9451780b568df24430dd0085ae951f097c#diff-22ebdb9efbd3783ad7fa874f3cdf6af17887c24d772ebb5bda6cbc5deb042152 added a comment to `onModifyRowHeight`:

https://github.com/handsontable/handsontable/blob/34793d5e9ee10176941ebe2589996daa80b5effd/src/plugins/hiddenRows/hiddenRows.js#L303-L311

Reading "Hook is triggered internally only for the visible rows" I would think that the `rowIndexMapper` takes care of hiding the rows more efficiently than setting a `0px` height. That is also based on the [8.0.0 release blog post](https://handsontable.com/blog/articles/2020/8/the-new-handsontable-8-is-now-available):

> When columns or rows are hidden, they are not physically rendered. This applies to several plugins that can “hide” rows and columns, namely: ...

### Context
I am migrating to 8.2.0 from 7. I previously hid rows using `modifyRow` (the first way described in https://handsontable.com/blog/articles/2016/3/3-ways-to-hide-rows-in-handsontable - it may be useful if that blog post received a warning message that it is not up to date for 8.x). I used `modifyRow` because `HiddenRows` was not fast enough for my data sets (several thousands of rows, with often many being hidden). I'm wondering whether I should be using `HiddenRows`, or add a `HidingMap` to `rowIndexMapper` myself.

### Your environment
* Handsontable version: master
* Browser Name and version: N/A
* Operating System: N/A

### LINKED ISSUE #8474: Hiding rows - misleading information
### Description
On the page we state that

> You can show/hide certain rows straight from the Context menu using the following keys: hidden_rows_show and hidden_rows_hide.

But the demo below this description uses the default context menu 

```js
data: Handsontable.helper.createSpreadsheetData(12,5),
  colHeaders: true,
  rowHeaders: true,
  contextMenu: true,
  hiddenRows: {
    rows: [3, 5, 9],
    indicators: true
}
```

Wouldn't it be better to use those defined keys or provide two demos,
* first with default demo to show that enabling hiding shows 'Hide row' / 'Show row'
* second with the  `hidden_rows_show` and `hidden_rows_hide` keys only

#### Other issues on the page

* the 'gets data' doesn't seem to sound good in 
> Rows being hidden are included in a DataMap (gets by the getData method), but they aren't rendered.

* remove one 'See the examples section for more details.' from the 'Additional options' paragraph.
Zrzut ekranu 2020-10-22 o 10 10 17

I would just leave the one at the end. They both redirect to the same place on the page.

### Related issue
* hidden columns - basically the same set of issues - https://github.com/handsontable/docs/issues/231

### Links
hiding rows - https://handsontable.com/docs/8.1.0/demo-hiding-rows.html
support task (origin) - https://github.com/handsontable/support/issues/259

### PR DESCRIPTION
**Pull request by kirszenbaum**
This PR:
- Fixes #7390 
- Fixes #8474 
- Rewrites the [Row hiding](https://handsontable.com/docs/row-hiding/) topic
- Rewrites both the `HiddenRows` and `HiddenColumns` API ref, to ensure consistency
- Polishes the [Column hiding](https://handsontable.com/docs/column-hiding/) topic (rewritten in #8606), to ensure consistency

[skip changelog]

### CODE REVIEW

File: docs/next/guides/rows/row-hiding.md [hunk 1]
```diff
@@ -9,77 +9,225 @@ canonicalUrl: /row-hiding
...
+Now, those rows are hidden by default:
+
+::: example #example2
 ```js
-const container = document.querySelector('#example1');
+const container = document.querySelector('#example2');
+
+const hot = new Handsontable(container, {
+  licenseKey: 'non-commercial-and-evaluation',
+  data: Handsontable.helper.createSpreadsheetData(12,5),
```
Comment (swistach) [inline docs/next/guides/rows/row-hiding.md:59]:
```suggestion
  data: Handsontable.helper.createSpreadsheetData(12, 5),
```

File: docs/next/guides/rows/row-hiding.md [hunk 2]
```diff
@@ -9,77 +9,225 @@ canonicalUrl: /row-hiding
...
+
+Enable both the [`ContextMenu`](@/api/contextMenu.md) plugin and the [`HiddenRows`](@/api/hiddenRows.md) plugin. Now, the [context menu](@/guides/accessories-and-menus/context-menu.md) automatically displays additional items for hiding and unhiding rows.
+
+::: example #example4
+```js
+const container = document.querySelector('#example4');
+
+const hot = new Handsontable(container, {
+  licenseKey: 'non-commercial-and-evaluation',
+  data: Handsontable.helper.createSpreadsheetData(12,5),
```
Comment (swistach) [inline docs/next/guides/rows/row-hiding.md:111]:
```suggestion
  data: Handsontable.helper.createSpreadsheetData(12, 5),
```

File: docs/next/guides/rows/row-hiding.md [hunk 3]
```diff
@@ -9,77 +9,225 @@ canonicalUrl: /row-hiding
...
 
-The plugin provides the option to display hidden row indicators in the headers to notify the user which rows have been hidden. To enable them, set the `indicators` property in the plugin's configuration object to `true`.
+::: example #example1
+```js
+const container = document.querySelector('#example1');
 
-To change the selection area of the copy/paste range, set the `copyPasteEnabled` property to `true` or `false`. By default, this property is set to `true`. If set to `false`, the hidden rows are being skipped for copy/paste actions.
+const hot = new Handsontable(container, {
+  licenseKey: 'non-commercial-and-evaluation',
+  data: Handsontable.helper.createSpreadsheetData(12,5),
```
Comment (swistach) [inline docs/next/guides/rows/row-hiding.md:32]:
```suggestion
  data: Handsontable.helper.createSpreadsheetData(12, 5),
```

File: docs/next/guides/rows/row-hiding.md [hunk 4]
```diff
@@ -9,77 +9,225 @@ canonicalUrl: /row-hiding
...
+If you use both the [`NestedHeaders`](@/api/nestedheaders.md) plugin and the [`HiddenRows`](@/api/hiddenRows.md) plugin, you also need to set the `colHeaders` property to `true`. Otherwise, `indicators` won't work.
+:::
+
+::: example #example3
+```js
+const container = document.querySelector('#example3');
 
 const hot = new Handsontable(container, {
+  licenseKey: 'non-commercial-and-evaluation',
   data: Handsontable.helper.createSpreadsheetData(12,5),
```
Comment (swistach) [inline docs/next/guides/rows/row-hiding.md:87]:
```suggestion
  data: Handsontable.helper.createSpreadsheetData(12, 5),
```

File: docs/next/guides/rows/row-hiding.md [hunk 5]
```diff
@@ -9,77 +9,225 @@ canonicalUrl: /row-hiding
...
+You can also add the row hiding menu items individually, by adding the [`hidden_rows_show`](@/guides/accessories-and-menus/context-menu.md#context-menu-with-specific-options) and [`hidden_rows_hide`](@/guides/accessories-and-menus/context-menu.md#context-menu-with-specific-options) strings to the `contextMenu` parameter:
 
-To access the plugin instance call:
+::: example #example5
+```js
+const container = document.querySelector('#example5');
+
+const hot = new Handsontable(container, {
+  licenseKey: 'non-commercial-and-evaluation',
+  data: Handsontable.helper.createSpreadsheetData(12,5),
```
Comment (swistach) [inline docs/next/guides/rows/row-hiding.md:134]:
```suggestion
  data: Handsontable.helper.createSpreadsheetData(12, 5),
```

File: docs/next/guides/rows/row-hiding.md [hunk 6]
```diff
@@ -9,77 +9,225 @@ canonicalUrl: /row-hiding
...
+
+To exclude hidden rows from copying and pasting, in the `hiddenRows` object, set the `copyPasteEnable` property to `false`:
+
+::: example #example6
+```js
+const container = document.querySelector('#example6');
+
+const hot = new Handsontable(container, {
+  licenseKey: 'non-commercial-and-evaluation',
+  data: Handsontable.helper.createSpreadsheetData(12,5),
```
Comment (swistach) [inline docs/next/guides/rows/row-hiding.md:159]:
```suggestion
  data: Handsontable.helper.createSpreadsheetData(12, 5),
```

File: docs/next/guides/rows/row-hiding.md [hunk 7]
```diff
@@ -9,77 +9,225 @@ canonicalUrl: /row-hiding
...
+  }
+});
+```
+:::
+
+## Row hiding API methods
+
+For the most popular row hiding tasks, use the API methods below.
+
+To see your changes, re-render your Handsontable instance with the [`hot.render()`](@/api/core.md#render) method.
```
Comment (swistach) [inline docs/next/guides/rows/row-hiding.md:177]:
```suggestion
To see your changes, re-render your Handsontable instance with the [`render()`](@/api/core.md#render) method.
```

File: docs/next/guides/columns/column-hiding.md [hunk 8]
```diff
@@ -42,31 +44,33 @@ To set up your column hiding configuration, follow the steps below.
...
+
+Now, those columns are hidden by default:
 
 ::: example #example2
 ```js
 const container = document.querySelector('#example2');
 
 const hot = new Handsontable(container, {
+  licenseKey: 'non-commercial-and-evaluation',
   data: Handsontable.helper.createSpreadsheetData(5,12),
```
Comment (swistach) [inline docs/next/guides/columns/column-hiding.md:59]:
```suggestion
  data: Handsontable.helper.createSpreadsheetData(5, 12),
```

File: docs/next/guides/columns/column-hiding.md [hunk 9]
```diff
@@ -79,40 +83,42 @@ If you use both the [`NestedHeaders`](@/api/nestedheaders.md) plugin and the [`H
 const container = document.querySelector('#example3');
 
 const hot = new Handsontable(container, {
+  licenseKey: 'non-commercial-and-evaluation',
   data: Handsontable.helper.createSpreadsheetData(5,12),
```
Comment (swistach) [inline docs/next/guides/columns/column-hiding.md:87]:
```suggestion
  data: Handsontable.helper.createSpreadsheetData(5, 12),
```

File: docs/next/guides/columns/column-hiding.md [hunk 10]
```diff
@@ -79,40 +83,42 @@ If you use both the [`NestedHeaders`](@/api/nestedheaders.md) plugin and the [`H
...
+
+Enable both the [`ContextMenu`](@/api/contextMenu.md) plugin and the [`HiddenColumns`](@/api/hiddenColumns.md) plugin. Now, the [context menu](@/guides/accessories-and-menus/context-menu.md) automatically displays additional items for hiding and unhiding columns.
 
 ::: example #example4
 ```js
 const container = document.querySelector('#example4');
 
 const hot = new Handsontable(container, {
+  licenseKey: 'non-commercial-and-evaluation',
   data: Handsontable.helper.createSpreadsheetData(5,12),
```
Comment (swistach) [inline docs/next/guides/columns/column-hiding.md:111]:
```suggestion
  data: Handsontable.helper.createSpreadsheetData(5, 12),
```

File: docs/next/guides/columns/column-hiding.md [hunk 11]
```diff
@@ -124,6 +130,7 @@ You can also add the column hiding menu items individually, by adding the [`hidd
 const container = document.querySelector('#example5');
 
 const hot = new Handsontable(container, {
+  licenseKey: 'non-commercial-and-evaluation',
   data: Handsontable.helper.createSpreadsheetData(5,12),
```
Comment (swistach) [inline docs/next/guides/columns/column-hiding.md:134]:
```suggestion
  data: Handsontable.helper.createSpreadsheetData(5, 12),
```

File: docs/next/guides/columns/column-hiding.md [hunk 12]
```diff
@@ -149,6 +155,7 @@ To exclude hidden columns from copying and pasting, in the `hiddenColumns` objec
 const container = document.querySelector('#example6');
 
 const hot = new Handsontable(container, {
+  licenseKey: 'non-commercial-and-evaluation',
   data: Handsontable.helper.createSpreadsheetData(5,12),
```
Comment (swistach) [inline docs/next/guides/columns/column-hiding.md:159]:
```suggestion
  data: Handsontable.helper.createSpreadsheetData(5, 12),
```

File: docs/next/guides/columns/column-hiding.md [hunk 13]
```diff
@@ -168,19 +174,19 @@ const hot = new Handsontable(container, {
 
 For the most popular column hiding tasks, use the API methods below.
 
-To see the changes that you made, re-render your table with the [`hot.render()`](@/api/core.md#render) method afterward.
+To see your changes, re-render your Handsontable instance with the [`hot.render()`](@/api/core.md#render) method.
```
Comment (swistach) [inline docs/next/guides/columns/column-hiding.md:177]:
```suggestion
To see your changes, re-render your Handsontable instance with the [`render()`](@/api/core.md#render) method.
```

**Review by swistach — CHANGES_REQUESTED** [review]
What do you think about omitting instances' references when we describe process/action using methods names?
Of course, an instance's reference is necessary for the code block.

**Comment by kirszenbaum** [comment]
> What do you think about omitting instances' references when we describe process/action using methods names?
> Of course, an instance's reference is necessary for the code block.

You mean e.g. `render()` rather than `hot.render()`? Yeah, definitely!

**Review by mrpiotr-dev — APPROVED** [review]
LGTM 👍

**Pull request merged**